packages/payment-detection/test/erc777/superfluid-retriever.test.ts

```diff
@@ -0,0 +1,122 @@
+import { SuperFluidInfoRetriever } from '../../src/erc777/superfluid-retriever';
+import PaymentReferenceCalculator from '../../src/payment-reference-calculator';
+import { utils } from 'ethers';
+import { PaymentTypes } from '@requestnetwork/types';
+
+describe('api/erc777/superfluid-info-retriever-1request', () => {
+  describe('on rinkeby', () => {
+    it('should get payment events from SuperFluid via subgraph with 1 request', async () => {
+      const paymentData = {
+        reference: '0xf81e1f6b6ff7cd15ab5538b403c8fdf3fed1f254ce6abcddf18941354039c752',
+        txHash: '0xe472ca1b52751b058fbdaeaffebd98c0cc43b45aa31794b3eb06834ede19f7be',
+        from: '0x9c040e2d6fd83a8b35069aa7154b69674961e0f7',
+        to: '0x52e5bcfa46393894afcfe6cd98a6761fa692c594',
+        network: 'rinkeby',
+        salt: '0ee84db293a752c6',
+        amount: '92592592592592000',
+        requestId: '0188791633ff0ec72a7dbdefb886d2db6cccfa98287320839c2f173c7a4e3ce7e1',
+        block: '9945543',
+        token: '0x745861aed1eee363b4aaa5f1994be40b1e05ff90',
+      };
+      const paymentReference = PaymentReferenceCalculator.calculate(
+        paymentData.requestId,
+        paymentData.salt,
+        paymentData.to,
+      );
+      const onChainReference = utils.keccak256(`0x${paymentReference}`);
+      expect(onChainReference).toEqual(paymentData.reference);
+
+      const graphRetriever = new SuperFluidInfoRetriever(
+        paymentReference,
+        paymentData.token,
+        paymentData.to,
+        PaymentTypes.EVENTS_NAMES.PAYMENT,
+        paymentData.network,
+      );
+      const transferEvents = await graphRetriever.getTransferEvents();
+      expect(transferEvents).toHaveLength(5);
+      expect(transferEvents[0].amount).toEqual(paymentData.amount);
+      expect(transferEvents[0].name).toEqual('payment');
+      expect(transferEvents[0].parameters?.to).toEqual(paymentData.to);
+      expect(transferEvents[1].amount).toEqual('34722222222222000');
+      expect(transferEvents[2].amount).toEqual('40509259259259000');
+      expect(transferEvents[0].parameters?.txHash).toEqual(paymentData.txHash);
+      expect(transferEvents[0].parameters?.block).toEqual(paymentData.block);
+    });
+  });
+});
+
+describe('api/erc777/superfluid-info-retriever-2requests', () => {
+  describe('on rinkeby', () => {
+    it('should get payment event from SuperFluid via subgraph with 2 requests', async () => {
+      const paymentData = {
+        reference: '0x23b12cbc2c59e39389f3c7cd652c4594a3a7595d7e9af265d07878345a6fe488',
+        from: '0x9c040e2d6fd83a8b35069aa7154b69674961e0f7',
+        to: '0x52e5bcfa46393894afcfe6cd98a6761fa692c594',
+        network: 'rinkeby',
+        salt: '0ee84db293a752c6',
+        amount: '320833333333331260',
```

Comment:
These tests rely on existing transactions so we should get more details about how to know why the expected amount is `320833333333331260`. Example:
```
        amount: '320833333333331260', // = (14 h 00 min 10 sec = 50'400 sec) x (1'000 DAI / sec)

Reply:
done for this most complex, it can be found for others at https://console.superfluid.finance/subgraph